### PR TITLE
DM-40247: Improve webdav interface as suggested by ruff S checks

### PR DESCRIPTION
**Pull request by airnandez**
## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

**Comment by timj** [comment]
To fix the mypy problem add something like:

```ini
[mypy-defusedxml.*]
ignore_missing_imports = True
```

This PR works with both xml and defusedxml so we do not have to wait on the rubin-env RFC.

Also, there is no reason why we can't add defusedxml as a requirement in the `pyproject.toml` and `requirements.txt` now for the pypi release and GitHub actions.

**Comment by airnandez** [comment]
Thanks @timj for your feedback. I added `defusedxml` to the appropriate files and now all the tests pass.

File: python/lsst/resources/http.py [hunk 1]
```diff
@@ -235,38 +265,88 @@ def _is_webdav_endpoint(path: ResourcePath | str) -> bool:
...
-                f"Unexpected response to OPTIONS request for {path}, status: {resp.status_code} {resp.reason}"
+        session = requests.Session()
+        session.mount(str(path), HTTPAdapter(max_retries=retries))
+        session.verify = os.environ.get("LSST_HTTP_CACERT_BUNDLE", True)
+        with session:
+            resp = session.options(
+                str(path),
+                stream=False,
+                timeout=(
+                    _timeout_from_environment("LSST_HTTP_TIMEOUT_CONNECT", 30.0),
```
Comment (timj) [inline python/lsst/resources/http.py:316]: Should we be reading the environment variables here if we are saying that OPTIONS should use a shorter timeout than other activities?

**Review by timj — APPROVED** [review]
Looks great. Thanks.

File: python/lsst/resources/http.py (hunk 1 above)
Comment (airnandez) [reply python/lsst/resources/http.py:316]: I thought using the timeout values specified by the user via the environment variables for `OPTIONS` as well would be better for consistency. In addition, using the environment allows the user to configure the environment for using a potential slow server, as opposed to hardcoding a magic value.

Comment (timj) [reply python/lsst/resources/http.py:316]: Yes, okay. I was concerned that people might want different values for OPTIONS and other things but if the env var is only used for testing it doesn't matter.

**Pull request merged**